### ISSUE TITLE
Feature Request: add distance markers to Party List HUD element.

### ISSUE THREAD

**Issue by CatbotTsumi** [PunishedPineapple/Distance#26]
As a healer i would love to know who is in range of my AOE heals! Trying to find them by nametags in world is hard when there's a lot going on or simply out of view. Having distance markers on the Party List would be amazing! 

This idea was partially inspired by another mod that puts the health % of enemies on the enmity list, so i figured something like that might be doable with the Party List and Distance mod.

**Comment by PunishedPineapple** [comment]
Distances on the party and enemy lists have been a planned feature for quite a while actually, but I've never gotten around to implementing them.  The goal was to have it either be distance readouts or let you color-code player/enemy name text based on distance.  There are a few things ahead of those features in the backlog, so I cannot really give a time estimate, though :(

**Comment by xyzsys** [comment]
Wow, I was just thinking about requesting this feature.

This would be nice to be able to select not just the distance from the player to other party member, but also the distance from the current target, the focus target, the closest member to the center of where the party members are, to each party member. 

It would also be easier to know who is closest to the center if there was a display option that would show "⭐️" for the member.

These could be used in a variety of ways, but especially would be helpful for healing, creating barriers, using buffs, etc.

**Comment by CatbotTsumi** [comment]
> Wow, I was just thinking about requesting this feature.
> 
> This would be nice to be able to select not just the distance from the player to other party member, but also the distance from the current target, the focus target, the closest member to the center of where the party members are, to each party member.
> 
> It would also be easier to know who is closest to the center if there was a display option that would show "⭐️" for the member.
> 
> These could be used in a variety of ways, but especially would be helpful for healing, creating barriers, using buffs, etc.

This sounds like a different feature request, you should probably open your own issue for it instead.

**Comment by PunishedPineapple** [comment]
I am not sure how to be able to display all of those distances without a lot of clutter in the party list, but I will try to think of a way to work them in.  Although, before any of that, I probably need to finally get around to implementing separate configuration profiles for each job I guess.

My initial thought is to have (in the plugin settings) a list where you can set a priority of which of the distances you mentioned to show (show from the soft-targeted player if you have one, otherwise from the targeted player, otherwise from any focus targeted player, otherwise from the player closest to the barycenter of all players, for example).  There would only be one distance readout per party member in the list.

The star by the name of the closest player is a really good idea, like the little triangle that shows when someone is below max HP.  I will definitely try to use that.

If you (or anyone else that cares) have other ideas on how to handle/present all that information, please let me know.

**Comment by xyzsys** [comment]
One idea is to display as values the distances from the player or from the temporary target/soft target to each party member, while the distance from the center could be displayed as a color-coded star itself.

This would allow to be displayed without disturbing the screen, although the exact distance from the starred member would not be known until the he becomes the target.

There are four different ranges for the healer's skill: 10y radius, 15y radius, 20y radius, and 30y radius

So it would be good if we have color-coded in 5 colors: 0-10y, 10-15y, 15-20y, 20-30y, and over 30y.

**Comment by xyzsys** [comment]
One idea is to display as values the distances from the player or from the temporary target/soft target to each party member, while the distance from the starred member could be displayed as a color-coded star itself.

This would allow to be displayed without disturbing the screen, although the exact distances from the starred member would not be known until that member becomes the target.

Skills that allow party members to be the center of the skill's range instead of the healer-self are:

Cure III 10y
Asylum 15y
Liturgy of the Bell 20y
Earthly Star 20y
Sacred Soil 15y
Deployment Tactics 30y

and faerie skills:

Whispering Dawn 15y
Fey Illumination 30y
Fey Blessing 20y
Angel's Whisper 15y
Seraphic Illumination 30y
Consolation 30y

Some of these actions actually ground target actions, but most gamepad players use these skills by targeting themselves or their party members with macros just as they would with regular skills.

in conclusion, there are four different ranges for the healer's skill: 10y radius, 15y radius, 20y radius, and 30y radius

So it would be good if we have color-coded in 5 colors: 0-10y, 10-15y, 15-20y, 20-30y, and over 30y.

Thinking about this, I feel that it would be better to have some kind of color-coded marks not only for the members closest to the center, but also for player-self and faerie to be able to indicate the distance range to the members.

**Comment by xyzsys** [comment]
I agree that some people do not need the exact distance values as you mentioned. Maybe there could be an option to hide the actual values.

My first thought was not to display everything at the same time, but to offer the user a variety of options and let them choose what to display and how to display it depending on their preference, purpose, and situation. I think this would make the feature usable in as many different use cases as possible.

**Comment by CatbotTsumi** [comment]
Like let them position and scale it and select which data(s) are displayed
sounds like a solid solution that should work for everyone.

**Comment by xyzsys** [comment]
For your reference:
https://en.wikipedia.org/wiki/Smallest-circle_problem
